### DISCUSSION TITLE
Show your dots!

### DISCUSSION THREAD

**Discussion by carloscabanero** [blinksh/blink#1523] [Show and tell]
What are your most used dotfiles? What makes them awesome? A cool zsh prompt? A tmux or screen trick? Share with us a link to a Gist or Repo, or just drop the interesting parts right here.

We are getting a new feature ready and would love to tweak things a bit based on your configurations. And pretty sure we will all learn something new too!

**Comment by agkozak** [comment]
[My dotfiles](https://github.com/agkozak/dotfiles) work on every OS and shell I ever have to use, but the `zsh`-specific scripts are the most advanced. They use my own [`zcomet`](https://github.com/agkozak/zcomet) plugin manager and my [`agkozak-zsh-prompt`](https://github.com/agkozak/agkozak-zsh-prompt).

I also have a well-developed [`.vimrc`](https://github.com/agkozak/dotfiles/blob/master/.vimrc).

Zenburn and Zenburn-like colors are used throughout.

**Reply by carloscabanero** [reply]
Thanks a lot for sharing, a lot of advanced stuff in there!

**Comment by Kudusch** [comment]
If we can also recommend other developer projects: [github.com/gpakosz/.tmux – Self-contained, pretty and versatile .tmux.conf configuration file.](https://github.com/gpakosz/.tmux)

**Reply by carloscabanero** [reply]
Never heard about it. Very cool

**Comment by pcho** [comment]
I have relatively simple and minimal dotfiles. I'm using ZSH as my daily driver https://github.com/pcho/zshfiles/ with a few aliases, added vim support, and custom completion options. For systems that don't need a lot of customizing, I use https://github.com/pcho/bashfiles.

For VIM, my editor of choice, I have another minimal setup: https://github.com/pcho/vimfiles/ supports TrueColor, custom bindings, a few plugins, and new addition, coc-nvim.

In the .dotfiles folder: https://github.com/pcho/dotfiles, I keep them all together, including config files for ssh and sshd that I use with my system setup script: https://github.com/pcho/binfiles/blob/master/bt. It helps me a lot with setting up new VPS when I needed or with daily tasks. When I used macOS, I also had this as helpers: https://github.com/pcho/binfiles/blob/master/.archive/setup-brew, to set up homebrew. And, https://github.com/pcho/binfiles/blob/master/.archive/setup-options-macos, for a bunch of options. 

For tmux, I use https://github.com/pcho/dotfiles/blob/master/.tmux.conf. It is for modern tmux versions, and I try to keep it minimal by removing defaults. Only one addition to it is https://github.com/christoomey/vim-tmux-navigator script. 

With this approach, I come to every software. I cleaned vimrc and zsh settings. I also have TrueColor support everywhere.

Everything is customized to work fine with the https://github.com/romainl/Apprentice color theme, which I slightly tweaked(by having darker background or status bar) in vim and terminal(Blink Shell theme):  https://gist.github.com/pcho/765a4cd56e59b72a7f1f971656eacb55.

**Reply by savagesmc** [reply]
I use zsh with tmux and vim/neovim. My dotfiles are [here](https://github.com/savagesmc/linux_config) which could be used by anyone, plus a personal customization [here](https://github.com/Savagesmc/custom_linux_config). For vi, I used the plugged plugin manager along with a number of vim plugins. I use this setup in linux, Darwin (Mac OS), windows subsystem for linux (WSL), and Cygwin.

**Comment by extratone** [comment]
I maintain a [public GitHub repo](https://github.com/extratone/blink) for Blink-specific configuration, if that's useful. It's not too exciting at the moment, but my [kb.json file](https://github.com/extratone/blink/blob/main/configuration/kb.json) as well as the contents of [/themes](https://github.com/extratone/blink/tree/main/themes) and [/fonts](https://github.com/extratone/blink/tree/main/fonts) can be manually installed to speed along a new user's experience, if they're interested.
